source/crud/crud.rst

```diff
@@ -1475,6 +1475,17 @@ Find And Modify
      */
     collation: Optional<Document>;
 
+    /**
+     * The index to use.
+     *
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * This option is only supported by servers >= 4.4. Older servers >= 4.2 will report an error for using this option.
+     * For servers < 4.2, the driver MUST raise an error if the caller explicitly provides a value.
```

Comment:
Note that this differs from #596. Strict checking of extra options in `findAndModify` was only introduced in 4.1.10 with [SERVER-40005](https://jira.mongodb.org/browse/SERVER-40005).

Reply:
It looks like there are only tests for servers <4.2 or >=4.4. Should we add a test against 4.2 to make sure that the driver does not raise a client side error (via a command started event)?

Reply:
I didn't think to do this since it felt like testing server behavior, but you make a good point that the APM expectation would allow us to assert that the driver isn't raising a client-side error.

I should point out that the language here is quite similar to `hint` for `update` (#596), and in the C driver implementation we opted to go beyond "For servers < 3.4" and raise client-side errors for any wire version before the options introduction since it matched what was already done for the `collation` and `arrayFilters` options. I suppose having a similar test as you're proposing for `hint` on `update` would fail in that case, but I'm happy to sort that out when/if such a test is added.

As for this PR, I can add a new test.

Reply:
I think one minor benefit for sending "hint" to 4.2 is that if the server backports "hint" support to 4.2 then drivers don't need to change.

Reply:
> I think one minor benefit for sending "hint" to 4.2 is that if the server backports "hint" support to 4.2 then drivers don't need to change.

Indeed. That did come up while discussing `update`.